src/Common/src/System/Net/Logging/NetEventSource.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+namespace System.Net
+{
+    [EventSource(Name = "Microsoft-System-Net",
+        Guid = "501a994a-eb63-5415-9af1-1b031260f16c")]
+    internal sealed class NetEventSource : EventSource
+    {
+        private const int FUNCTIONSTART_ID = 1;
+        private const int FUNCTIONSTOP_ID = 2;
+        private const int CRITICALEXCEPTION_ID = 3;
+        private const int CRITICALERROR_ID = 4;
+
+        private readonly static NetEventSource s_log = new NetEventSource();
+        private NetEventSource() { }
+        public static NetEventSource Log
+        {
+            get
+            {
+                return s_log;
+            }
+        }
+
+        [NonEvent]
+        internal static void Enter(ComponentType componentType, object obj, string method, object paramObject)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            string callerName;
+            int callerHash;
+            string parametersName = "";
+            int parametersHash = 0;
+            if (obj is string)
+            {
+                callerName = obj as string;
+                callerHash = 0;
+            }
+            else
+            {
+                callerName = LoggingHash.GetObjectName(obj);
+                callerHash = LoggingHash.HashInt(obj);
+            }
+
+            if (paramObject is string)
+            {
+                parametersName = paramObject as string;
+                parametersHash = 0;
+            }
+
+            else if (paramObject != null)
+            {
+                parametersName = LoggingHash.GetObjectName(paramObject);
+                parametersHash = LoggingHash.HashInt(paramObject);
+            }
+            s_log.FunctionStart(callerName,
+                                callerHash,
+                                LoggingHash.GetObjectName(method),
+                                parametersName,
+                                parametersHash,
+                                componentType);
+        }
+
+        [Event(FUNCTIONSTART_ID, Keywords = Keywords.FunctionEntryExit,
+    Level = EventLevel.Verbose, Message = "[{5}] {0}#{1}::{2}({3}#{4})")]
```

Comment:
Nit: alignment. (Also, our standard is to separate each argument on a new line.)